pandas/core/indexes/datetimes.py

```diff
@@ -1265,14 +1265,32 @@ def _parsed_string_to_bounds(self, reso, parsed):
         ----------
         reso : Resolution
             Resolution provided by parsed string.
-        parsed : datetime
+        parsed : datetime | object
             Datetime from parsed string.
 
         Returns
         -------
         lower, upper: pd.Timestamp
 
         """
+        if not hasattr(parsed, "tzinfo"):
```

Comment:
wrap parsed in ``Timestamp``, you are using datetime routines which may not always be applicable depending on tz's are used (e.g. if dateutil or pytz).

Reply:
can you please elaborate on that?

Reply:
I assume datetime objects here, so I expect the timezones to be datetime compatible. How could I wrap them in a way that this always works? Could you sketch some test cases I need to satisfy? Thanks a lot!

Reply:
``parsed = Timestamp(parsed)``

then you only care about ``.tz`` and can do ``.tz_convert()`` and such, much more idiomatic. we rareley deal directly with ``datetime`` objects